lib/logflare_web/controllers/plugs/compressed_body_reader.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Plug.Parsers.CompressedBodyReader do
+  def read_body(conn, opts \\ []) do
+    content_encoding = Plug.Conn.get_req_header(conn, "content-encoding")
+
+    with {:ok, body, conn} <- Plug.Conn.read_body(conn, opts) do
+      case try_decompress(body, content_encoding) do
+        {:ok, data} -> {:ok, data, conn}
+        {:more, data} -> {:more, data, conn}
+        {:error, _} = error -> error
+      end
+    end
+  end
+
+  defp try_decompress(data, []), do: {:ok, data}
+  defp try_decompress(data, ["gzip"]), do: safe_gunzip(data)
+
+  defp safe_gunzip(data) do
+    z = :zlib.open()
+    try do
+      :zlib.inflateInit(z, 31)
+      result = :zlib.safeInflate(z, data)
```

Comment:
from my understanding, the safeInflate throw an exception if memory exceeds, which would not result in error tuple above at line 9

Reply:
not really an issue since it would 500. Would be good to handle it with a 400 bad request though.

Reply:
It will not, it will then return `{:continue, deflated}` and you need to call it again with new data. So it will never throw. And even if, we safely handle that there and it will either return 413 (most cases) or it will throw 500 (less common case).